Added basic photo capability.

ProCynic opened the pull request:
New routes:
    GET     /photos      #view all photos
    POST    /photos      #add a photo with logged in user as owner
    GET     /photos/id   #view a specific photo
    DELETE  /photos/id   #delete a photo
    GET     /users/id/photos #get all photos for a specific user

New model Photo.java has image blob and User owner.

New view template photos.html.  Basic skeleton, needs work.

New Application.java methods:
    photos(Long id) //view all photos, or all photos by a specific user
    getPhoto(Long id)   //render a specific photo
    addPhoto(Photo photo) // add a photo with the logged in user as owner
    removePhoto(Long id) // delete a photo

ProCynic committed:
Added basic photo capability.

New routes:
    GET		/photos		 #view all photos
    POST	/photos		 #add a photo with logged in user as owner
    GET		/photos/id	 #view a specific photo
    DELETE	/photos/id	 #delete a photo
    GET		/users/id/photos #get all photos for a specific user

New model Photo.java has image blob and User owner.

New view template photos.html.  Basic skeleton, needs work.

New Application.java methods:
    photos(Long id)	//view all photos, or all photos by a specific user
    getPhoto(Long id) 	//render a specific photo
    addPhoto(Photo photo) // add a photo with the logged in user as owner
    removePhoto(Long id) // delete a photo

ProCynic committed:
Fixed indentation / tabs in Photo.java and conf/routes

ProCynic commented on conf/routes:
```diff
@@ -11,6 +11,13 @@ GET     /                                       Application.news
+GET     /photos                                 Application.photos(id:null)
+POST    /photos                                 Application.addPhoto
+GET     /photos/{<[0-9]+>id}                    Application.getPhoto
+DELETE  /photos/{<[0-9]+>id}                    Application.removePhoto
```
Needs authentication.

ProCynic committed:
added .project to .gitignore

amshali commented on app/models/Photo.java:
```diff
@@ -0,0 +1,13 @@
+package models;
```
Please add dateAdded and caption to this entity.

amshali commented on app/controllers/Application.java:
```diff
@@ -177,9 +177,39 @@ public static void deleteComment(Long id, Long userId) {
     c.delete();
     news(userId);
   }
-
```
Let's refactor the functionalities  for photo  into a different class in controller.

amshali commented on app/controllers/Application.java:
```diff
@@ -177,9 +177,39 @@ public static void deleteComment(Long id, Long userId) {
   }
-}
+
+  public static void photos(Long id) {
```
Please change the name of id parameter to ownerId or userId.

amshali commented on app/controllers/Application.java:
```diff
@@ -177,9 +177,39 @@ public static void deleteComment(Long id, Long userId) {
+    render(photos);
+  }
+  
+  public static void getPhoto(Long id) {
```
Change the name of parameter

ProCynic committed:
Fixes to photos:

  refactored photo controllers to controllers/Photos.java
  moved /views/Application/photos.html to /views/Photos/photos.html
  changed routes to reflect refactoring

  changed variable names for photo routes/controllers to be more descriptive

  app now returns 404 for bad photo requests

  cannot upload blank photo

ProCynic committed:
fixed unresolved merge conflict in Application.java

ProCynic commented on app/controllers/Photos.java:
```diff
@@ -0,0 +1,68 @@
+  }
+
+  public static void addPhoto(Photo photo) {
+    if (photo.image == null);{
```
BUG!  ; after if statement on this line.

amshali replied:
Could you please send a revision?

-- Amin

On Mon, Feb 27, 2012 at 5:12 PM, Geoffrey Parker <
reply@reply.github.com

> wrote:
> 
> > -  }
> >   +
> > -  public static void getPhoto(Long photoId) {
> > -    Photo photo = Photo.findById(photoId);
> > -    if (photo == null) {
> > -      response.status = Http.StatusCode.NOT_FOUND;
> > -      renderText("");
> > -    }
> > -    else {
> > -      response.setContentTypeIfNotSet(photo.image.type());
> > -      renderBinary(photo.image.get());
> > -    }
> > -  }
> >   +
> > -  public static void addPhoto(Photo photo) {
> > -    if (photo.image == null);{
> 
> BUG!  ; after if statement on this line.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/w7cook/OpenBook/pull/10/files#r493596

ProCynic replied:
there you go.
-Geoffrey Parker

On Mon, Feb 27, 2012 at 5:17 PM, amshali <
reply@reply.github.com

> wrote:
> 
> > -  }
> >   +
> > -  public static void getPhoto(Long photoId) {
> > -    Photo photo = Photo.findById(photoId);
> > -    if (photo == null) {
> > -      response.status = Http.StatusCode.NOT_FOUND;
> > -      renderText("");
> > -    }
> > -    else {
> > -      response.setContentTypeIfNotSet(photo.image.type());
> > -      renderBinary(photo.image.get());
> > -    }
> > -  }
> >   +
> > -  public static void addPhoto(Photo photo) {
> > -    if (photo.image == null);{
> 
> Could you please send a revision?
> 
> -- Amin
> 
> On Mon, Feb 27, 2012 at 5:12 PM, Geoffrey Parker <
> reply@reply.github.com
> 
> > wrote:
> > 
> > > -  }
> > >   +
> > > -  public static void getPhoto(Long photoId) {
> > > -    Photo photo = Photo.findById(photoId);
> > > -    if (photo == null) {
> > > -      response.status = Http.StatusCode.NOT_FOUND;
> > > -      renderText("");
> > > -    }
> > > -    else {
> > > -      response.setContentTypeIfNotSet(photo.image.type());
> > > -      renderBinary(photo.image.get());
> > > -    }
> > > -  }
> > >   +
> > > -  public static void addPhoto(Photo photo) {
> > > -    if (photo.image == null);{
> > 
> > BUG!  ; after if statement on this line.
> > 
> > ---
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/w7cook/OpenBook/pull/10/files#r493596
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/w7cook/OpenBook/pull/10/files#r493628

ProCynic committed:
Minor changes an bug fixes:

      fixed addPhoto bug with misplaced semicolon
      fixed favicon.ico route // still think this needs to be changed

      changed addPhoto to redirect to logged in user's photo page
      changed getPhoto to use Application.notFound() on 404

ProCynic commented on app/models/Photo.java:
```diff
@@ -0,0 +1,23 @@
+  public Blob image;
+
+  @Required
+  public Date postedAt;
```
dateAdded

ProCynic commented on app/models/Photo.java:
```diff
@@ -0,0 +1,23 @@
+  @Required
+  public Date postedAt;
+
+  public String caption;
```
caption
